script.js

```diff
@@ -1 +1,282 @@
+      // By default, the card name is the same as rankCounter
+      let cardName = `${rankCounter}`;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName === "1") {
+        cardName = "ace";
+      } else if (cardName === "11") {
+        cardName = "jack";
+      } else if (cardName === "12") {
+        cardName = "queen";
+      } else if (cardName === "13") {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      const card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        suitIcon: currentSuitIcon,
+      };
+
+      // Add the new card to the deck
+      newDeck.push(card); // add double the cards to the deck
+      newDeck.push(card);
+    }
+  }
+
+  // Return the completed card deck
+  return newDeck;
+};
+
+const squareClick = (cardElement, column, row) => {
+
+  //start the timer only when it is new round
+  if (isNewRound){
+  displayTimer();
+  
+  //remove all board elements once 3 mins is up
+  setTimeout(() => boardEl.innerText = 'Oh no, 3 mins is up! You didnt complete the game in time :(', 1000*3*60);
+
+  isNewRound = false;
+  }
+
+  console.log(cardElement);
+
+  console.log("FIRST CARD DOM ELEMENT", firstCard);
+
+  console.log("BOARD CLICKED CARD", board[column][row]);
+
+  const clickedCard = board[column][row];
+
+  // the user already clicked on this square
+  if (cardElement.innerText !== "") {
+    return;
+  }
+
+  // first turn
+  if (firstCard === null) {
+    console.log("first turn");
+    stateElement.innerText = 'You clicked your first card. Please click your second card.'
+    firstCard = clickedCard;
+    // turn this card over
+    cardElement.innerText = `${firstCard.name} ${firstCard.suitIcon}`;
+
+    // hold onto this for later when it may not match
+    firstCardElement = cardElement;
+
+    // second turn
+  } else {
+    console.log("second turn");
+    stateElement.innerText = 'You clicked your second card.';
+    if (
+      clickedCard.name === firstCard.name &&
+      clickedCard.suit === firstCard.suit
+    ) {
+      console.log("match");
+    
+      stateElement.innerText = 'You clicked your second card. Yay, it was a match!';
+
+      //Remove the match message after 3 seconds
+      setTimeout(() => {stateElement.innerText = 'Click another card to start matching again.';}, 3000);
+
+      // turn this card over
+      cardElement.innerText = `${clickedCard.name} ${clickedCard.suitIcon}`;
+    } else {
+      console.log("NOT a match");
+      stateElement.innerText += ' Oh no, it was not a match! Click another card to restart the matching.';
+
+      //show both cards first
+      firstCardElement.innerText = `${firstCard.name} ${firstCard.suitIcon}`;
+      cardElement.innerText = `${clickedCard.name} ${clickedCard.suitIcon}`;
+
+      //show cards for only 3 seconds, then turn both cards over
```

Comment:
When it is not a match, use a `canClick` global variable to make the player unable to click on another card while waiting for the 2 cards' innerText to be cleared. If not, when the player clicks on a 3rd card while the 2 cards are still faced up, the 1st face up card will not disappear after 3 seconds because the 1st card is now the 3rd card due to line 158.